### ISSUE TITLE
Problem about training

### ISSUE THREAD

**Issue by 1563245379** [xuyige/SoftCoT#21]
When I try to train the projector using the command below, the results I get on gsm8k are significantly lower than those obtained by testing with the released weights  (only about 76%). I want to know whether the final saved weights are from the last epoch or from the best epoch on the validation set.

CUDA_VISIBLE_DEVICES=0 python train_softcot.py \
    --large_model_id Qwen/Qwen2.5-7B-Instruct \
    --small_model_id Qwen/Qwen2.5-1.5B-Instruct \
    --output_name [Output Name] \
    --batch_size 4 \
    --task_name gsm8k \
    --num_thought_tokens 32 \
    --n_epochs 10 \

**Comment by xuyige** [comment]
Thanks for your interest to our work. The final saved weights are from the last epoch. Hope it is helpful to you.

**Comment by Mrwang8633** [comment]
> When I try to train the projector using the command below, the results I get on gsm8k are significantly lower than those obtained by testing with the released weights (only about 76%). I want to know whether the final saved weights are from the last epoch or from the best epoch on the validation set.当我尝试使用以下命令训练投影头时，在 gsm8k 数据集上得到的结果显著低于使用发布权重测试得到的结果（仅约 76%）。我想知道最终保存的权重是来自最后一个 epoch 还是验证集上表现最佳的 epoch。
> 
> CUDA_VISIBLE_DEVICES=0 python train_softcot.py CUDA_VISIBLE_DEVICES=0 python train_softcot.py --large_model_id Qwen/Qwen2.5-7B-Instruct --large_model_id Qwen/Qwen2.5-7B-Instruct --small_model_id Qwen/Qwen2.5-1.5B-Instruct --small_model_id Qwen/Qwen2.5-1.5B-Instruct --output_name [Output Name] --output_name [输出名称] --batch_size 4 --batch_size 4 --task_name gsm8k --任务名称 gsm8k --num_thought_tokens 32 --num_thought_tokens 32 --n_epochs 10 \ --n_epochs10\

I cannot even reach the 76% accuracy. Could I ask for your advice? My email address is:wangyimeng@hrbeu.edu.cn

**Comment by xuyige** [comment]
For training details, please refer to README.md :

> 17-May-2025: To help reproducing, we release some useful data as well as an official reproduce of SoftCoT, check the data [here](https://huggingface.co/datasets/xuyige/ASDiv-Aug), check the logs and checkpoints [here](https://drive.google.com/file/d/1rC8x0dFv2oCEg_dCBOqd0lzGUwoNdLCY/view?usp=drive_link).

If you have any other problems, please feel free to contact me.
